include/message_serialization/binary_serialization.h

```diff
@@ -41,52 +39,88 @@ inline bool serializeToBinary(const std::string& file,
   std::ofstream ofs(file, std::ios::out|std::ios::binary);
   if(ofs)
   {
-    ofs.write((char*) buffer.get(), serial_size);
-    return ofs.good();
+      ofs.write((char*) buffer.get(), serial_size);
+      if(!ofs.good())
+        throw std::runtime_error("Failed to write to binary file stream at '" + file + "'");
   }
 
-  return false;
+  throw std::runtime_error("Failed to open binary file stream at '" + file + "'");
```

Comment:
This function seems like it has a problem. If the file does not open "!ofs" it throws. If after a write, "!ofs.good()" it throws. This is all good however if "ofs.good()" then it falls through and throws. I think you need a return.

Reply:
Good catch. I just fixed this